tool/enc-unicode.rb

```diff
@@ -166,9 +166,9 @@ def parse_scripts(data, categories)
         categories[current] = file[:title]
         (names[file[:title]] ||= []) << current
         cps = []
-      elsif /^(\h+)(?:\.\.(\h+))?\s*;\s*(\w+)/ =~ line
-        current = $3
+      elsif /^(\h+)(?:\.\.(\h+))?\s*;\s*(\w(?>[\w\s;]*\w)?)/ =~ line
         $2 ? cps.concat(($1.to_i(16)..$2.to_i(16)).to_a) : cps.push($1.to_i(16))
+        current = $3.gsub(/\W+/, '_')
```

Comment:
Could you add a comment that shows an example matched line and `$1`/`$2`/`$3` for the case for easy to understand?

Reply:
I have added some comments. https://github.com/ruby/ruby/pull/12798/files#diff-66b721e4f3da90f52f789b006bac8743157852b8bea2643ce866d70296acaec9R171-R179